server/resourcemanager/resource_manager.go

```diff
@@ -32,55 +32,109 @@ type Resource[T ResourceSpec] struct {
 	Spec T      `mapstructure:"spec"`
 }
 
-type SortableHandler interface {
-	SortingFields() []string
-}
-
-type ResourceHandler[T ResourceSpec] interface {
-	SortableHandler
-	SetID(T, id.ID) T
-	List(_ context.Context, take, skip int, query, sortBy, sortDirection string) ([]T, error)
-	Count(_ context.Context, query string) (int, error)
-	Create(context.Context, T) (T, error)
-	Update(context.Context, T) (T, error)
-	Get(context.Context, id.ID) (T, error)
-	Delete(context.Context, id.ID) error
+type Manager interface {
+	EnabledOperations() []Operation
+	Handler() any
+	RegisterRoutes(*mux.Router) *mux.Router
 }
 
 type manager[T ResourceSpec] struct {
 	resourceType string
-	handler      ResourceHandler[T]
-	idgen        func() id.ID
+	handler      any
+	rh           resourceHandler[T]
+	config       config
+}
+
+type config struct {
+	enabledOperations []Operation
+	idgen             func() id.ID
+}
+
+type managerOption func(*config)
+
+func WithIDGen(fn func() id.ID) managerOption {
+	return func(c *config) {
+		c.idgen = fn
+	}
 }
 
-func New[T ResourceSpec](resourceType string, handler ResourceHandler[T], idgenFn func() id.ID) *manager[T] {
+func WithOperations(ops ...Operation) managerOption {
+	return func(c *config) {
+		c.enabledOperations = ops
+	}
+}
+
+func New[T ResourceSpec](resourceType string, handler any, opts ...managerOption) Manager {
+	rh := &resourceHandler[T]{}
+
+	cfg := config{
+		enabledOperations: availableOperations,
+		idgen:             func() id.ID { return id.ID("") },
+	}
+
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	err := rh.bindOperations(cfg.enabledOperations, handler)
```

Comment:
You may be wandering: Why do we need to provide an explicit list of `enabledOperations` if we can infer them from the given `handler`?
You might notice that the `bindOperations` actually checks tha the `handler` has the requried methods and "binds" them to the internal `resourceHandler`. Why don't just rely on that?

The answer is, to double check. This applies mainly to the case when a developer is creating a new, full CRUD resource. If we just rely on the information we can infer, it could happen that a dev forgets to implement the `create` method in the repository, and that would make the ResourceManager to *not* "activate" the create endpoint.

So, by forcing developers to be explicit about resources not being fully CRUD, we have an extra validation that the implemented behaviour is the expected one.